scout/server/blueprints/cases/controllers.py

```diff
@@ -198,6 +212,39 @@ def case(store, institute_obj, case_obj):
     return data
 
 
+def _check_outdated_gene_panel(panel_obj, latest_panel):
+    """Compare genes of a case gene panel with the latest panel version and return differences
+
+    Args:
+        panel_obj(dict): the gene panel of a case
+        latest_panel(dict): the latest version of that gene panel
+
+    returns:
+        missing_genes, extra_genes
+    """
+    # Create a list of minified gene object for the case panel {hgnc_id, gene_symbol}
+    hgnc_ids_case_panel = [
+        {"hgnc_id": gene["hgnc_id"], "symbol": gene.get("symbol", gene["hgnc_id"])}
+        for gene in panel_obj["genes"]
+    ]
+    # And for the latest panel
+
+    hgnc_ids_latest_panel = [
+        {"hgnc_id": gene["hgnc_id"], "symbol": gene.get("symbol", gene["hgnc_id"])}
+        for gene in latest_panel["genes"]
+    ]
+    # Extract the genes unique to case panel
+    extra_genes = [
+        gene["symbol"] for gene in hgnc_ids_case_panel if gene not in hgnc_ids_latest_panel
```

Comment:
Sets might have been even faster, but this doesn't seem bad either. Nicely done!

Reply:
> @dnil should be fixed now. It started complaining for the code complexity, so I moved the gene check to a separate function. It didn't work. I think it keeps complaining because the case controller is still very long but it was like this from before..

Well, yes, I agree it is complex, and I suppose what pushed it over the edge is the new nested if statement. Its possible you would get away with it if you just un-nest them - the vars will not be more or less set if you check them outside the outer if-statement I suppose.

Reply:
That is to say, I don't mind much either way, being c-schooled, but I suppose flat if-statements is more the python way.

Reply:
> Sets might have been even faster, but this doesn't seem bad either. Nicely done!

The problem with sets here is that I'm comparing the hgnc ids, but I need to keep also the symbols, and to do so I store the values in dictionaries. Dictionaries can't be stored in sets (not hashable) so I'd need to have two loops instead, and in the end it would be slower I think. I'll keep it like this instead

Reply:
> That is to say, I don't mind much either way, being c-schooled, but I suppose flat if-statements is more the python way.

I wouldn't like to add even more code to a function that by itself is simple. It's just that it's part of a nested check and the complexity is also given by the levels above. Let's keep it this way. We can fix it in a second moment if we feel like it.